### PR TITLE
feat(core): add iterator support and assign methods to FixedVector

### PR DESCRIPTION
**Pull request by dkrutskikh**
### **PR Type**
Enhancement

___

### **Description**
- Add iterator type aliases and range-based constructor

- Implement three overloaded `assign()` methods for flexible content replacement

- Standardize size type usage with `size_type` typedef throughout

- Add comprehensive test coverage for new iterator and assign functionality

___

### Diagram Walkthrough

```mermaid
flowchart LR
  A["FixedVector"] -->|"add iterator types"| B["iterator aliases"]
  A -->|"add constructor"| C["range constructor"]
  A -->|"add methods"| D["assign overloads"]
  C -->|"uses"| B
  D -->|"uses"| B
  E["Tests"] -->|"validate"| C
  E -->|"validate"| D
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>fixed_vector.hpp</strong><dd><code>Add iterator types and assign methods</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

include/core/fixed_vector.hpp

<ul><li>Add four iterator type aliases: <code>iterator</code>, <code>const_iterator</code>, <br><code>reverse_iterator</code>, <code>const_reverse_iterator</code><br> <li> Add range-based constructor template accepting <code>InputIterator</code> with <br>SFINAE guard<br> <li> Add three <code>assign()</code> method overloads: count/value, iterator range, and <br>initializer list<br> <li> Standardize all <code>std::size_t</code> occurrences to <code>size_type</code> typedef<br> <li> Update parameter name from <code>init</code> to <code>ilist</code> in initializer list operator <br>for consistency</ul>

</details>

  </td>
  <td><a href="https://github.com/ToymanInteractive/toygine2/pull/175/files#diff-5b1f34cbe405ace7840f20236175cd524c075fa02158cfd9453895983780a9e6">+91/-8</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>fixed_vector.inl</strong><dd><code>Implement iterator constructor and assign methods</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

include/core/fixed_vector.inl

<ul><li>Implement range-based constructor using iterator pair with distance <br>validation<br> <li> Implement three <code>assign()</code> method templates with proper element clearing <br>and construction<br> <li> Replace all <code>std::size_t</code> loop variables with <code>size_type</code> for consistency<br> <li> Update initializer list constructor to use <code>auto</code> for index variable<br> <li> Add assertions to validate iterator range sizes against capacity</ul>

</details>

  </td>
  <td><a href="https://github.com/ToymanInteractive/toygine2/pull/175/files#diff-3ac0b9698dbeabfb53240d9d532e20e690cc4830b529c9cc74b621295273c35e">+61/-12</a>&nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>fixed_vector_test.cpp</strong><dd><code>Add iterator and assign method tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

test/core/fixed_vector_test.cpp

<ul><li>Add four test sections for iterator constructor with vectors, arrays, <br>empty ranges, and custom types<br> <li> Add comprehensive test suite for <code>assign()</code> methods covering <br>count/value, iterator ranges, initializer lists<br> <li> Test edge cases including empty ranges, zero counts, maximum capacity, <br>and chained operations<br> <li> Validate behavior with complex types like nested vectors and custom <br>string types</ul>

</details>

  </td>
  <td><a href="https://github.com/ToymanInteractive/toygine2/pull/175/files#diff-23569edf61c9bda055b16e8ede3e8b7bd2e1f61f777b02545d878fb913a2dccd">+242/-0</a>&nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Configuration changes</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>settings.json</strong><dd><code>Add cstddef to VSCode settings</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

.vscode/settings.json

- Add `cstddef` to C++ language association for proper IDE support

</details>

  </td>
  <td><a href="https://github.com/ToymanInteractive/toygine2/pull/175/files#diff-a5de3e5871ffcc383a2294845bd3df25d3eeff6c29ad46e3a396577c413bf357">+2/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

</details>

___

### CODE REVIEW

File: include/core/fixed_vector.hpp [hunk 1]
```diff
@@ -254,25 +277,85 @@ class FixedVector {
...
-    \post This vector has the same size and content as the source \a init.
+    \post This vector has the same size and content as the source \a ilist.
     \post All elements are properly copied using copy construction.
     \post All previous elements are properly destroyed.
 
     \note The operator handles empty initializer lists correctly.
   */
-  FixedVector<type, allocatedSize> & operator=(std::initializer_list<type> init) noexcept;
+  FixedVector<type, allocatedSize> & operator=(std::initializer_list<type> ilist) noexcept;
+
```
[the remark this answers is not in the document]
Comment (dkrutskikh) [reply include/core/fixed_vector.hpp]: fixed

File: include/core/fixed_vector.inl [hunk 2]
```diff
@@ -187,13 +199,50 @@ inline FixedVector<type, allocatedSize> & FixedVector<type, allocatedSize>::oper
...
+inline void FixedVector<type, allocatedSize>::assign(InputIterator first, InputIterator last) noexcept {
+  assert_message(std::distance(first, last) <= allocatedSize, "Iterator range size must not exceed capacity.");
+
+  clear();
+
+  for (auto it = first; it != last; ++it) {
+    push_back(*it);
+  }
+}
+
```
[the remark this answers is not in the document]
Comment (dkrutskikh) [reply include/core/fixed_vector.inl]: fixed
